faq.md

```diff
@@ -499,6 +499,90 @@ If you are using the Flutter plugin for IntelliJ, you can launch the app in
 profile or release mode using the menu entries **Run > Flutter Run in Profile Mode**
 or **Release Mode**.
 
+### What programming paradigm does Flutter's framework use?
+
+Flutter is a multi-paradigm programming environment. Many programming techniques
+developed over the past few decades are used in Flutter, each to solve
+a particular set of problems in a clean way. In no particular order:
+
+* Composition: The primary paradigm used by Flutter is that of using small objects with narrow
+scopes of behavior, composed together to obtain more complicated effects. Most widgets
+in the Flutter widget library are built in this way.
+
+* Functional programming: Entire applications can be built with only
+[StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html)s,
+which are essentially functions that describe how arguments map to other functions,
```

Comment:
What do you mean "how arguments map to other functions"? Are you mapping arguments to functions, or are you just calling other functions using arguments that have been mapped?

Reply:
a StatelessWidget is literally equivalent to a function with the signature `(constructor arguments) -> Widget`. If you build your app exclusively with StatelessWidgets, then you are mapping the constructor arguments to more StatelessWidgets, which as previously established are equivalent to functions, so they are "essentially functions that describe how arguments map to other functions".